### PR TITLE
Multiple updates of raptor studio

### PR DESCRIPTION
**Pull request by bebef1987**

### CODE REVIEW

File: apps/android/firefox.py [hunk 1]
```diff
@@ -54,11 +55,20 @@ def create_certificate(self):
 
     def setup_device(self):
         self.device = ADBAndroid()
-        if self.binary:
-            if self.device.is_app_installed(self.APP_NAME):
-                print("Uninstalling app %s" % self.APP_NAME)
-                self.device.uninstall_app(self.APP_NAME)
-            self.device.install_app(apk_path=self.binary)
+        if self.binary and self.proxy.mode is "record":
+            if not self.app_installed:
```
Comment (davehunt) [inline apps/android/firefox.py]: This variable name makes the code difficult to read as we're first checking if the app is installed and then checking it again by querying the device. Perhaps rename it to `skip_install`?

File: proxy/mitmproxy.py [hunk 2]
```diff
@@ -37,8 +37,8 @@ def start(self):
 
         return self.process
 
-    def stop(self):
-        if self.mode is "record":
+    def stop(self, kill=True):
```
Comment (davehunt) [inline proxy/mitmproxy.py]: Why is the optional kwarg added here? I don't see where we pass anything to this method.

File: proxy/mitmproxy.py [hunk 3]
```diff
@@ -37,8 +37,8 @@ def start(self):
...
+        if kill:
             # Record mode. Send proxy stop command and wait for it to close.
```
Comment (davehunt) [inline proxy/mitmproxy.py]: This comment no longer makes sense if we stop the proxy in replay mode too.

File: studio.py [hunk 4]
```diff
@@ -61,6 +61,19 @@ def _digest_file(self, file, algorithm):
             print("hashed %s with %s to be %s" % (name, algorithm, h.hexdigest()))
             return h.hexdigest()
 
+    def replay_site(self, site, screen_shot=False):
```
Comment (davehunt) [inline studio.py]: This appears to mostly duplicate `replaying`. Can we reduce the duplication? If replaying currently waits for the user to kill the process, maybe we should change it to the 5 second sleep and screenshot by default instead.

File: studio.py [hunk 5]
```diff
@@ -166,6 +189,15 @@ def update_json_information(self, site):
...
             json_data = json.loads(f.read())
 
+        # Generate the eTLD+1 list
+        # Bug 1585598 - Validate list of sites used for testing Fission
+        etdl = []
+        for item in json_data['http_protocol'].keys():
+            base_url = ".".join(item.split('.')[-2:])
+            if base_url not in etdl:
+                etdl.append(base_url)
+        self.information["etdl"] = etdl
```
Comment (davehunt) [inline studio.py:202]: Nice!

File: studio.py [hunk 6]
```diff
@@ -175,7 +207,7 @@ def update_json_information(self, site):
 
         json_data["info"] = self.information
         with open(site["json_path"], "w") as f:
-            f.write(json.dumps(json_data))
+            f.write(json.dumps(json_data, sort_keys=True, indent=2))
```
Comment (davehunt) [inline studio.py:213]: 👍

**Review by davehunt — CHANGES_REQUESTED** [review]

File: apps/android/firefox.py (hunk 1 above)
Comment (bebef1987) [reply apps/android/firefox.py]: renamed

File: proxy/mitmproxy.py (hunk 3 above)
Comment (bebef1987) [reply proxy/mitmproxy.py]: removed the `Record mode.`

File: proxy/mitmproxy.py (hunk 2 above)
Comment (bebef1987) [reply proxy/mitmproxy.py]: removed the argument and updated the stop function

File: studio.py (hunk 4 above)
Comment (bebef1987) [reply studio.py]: Updated to run the same method

File: studio.py [hunk 7]
```diff
@@ -61,17 +61,35 @@ def _digest_file(self, file, algorithm):
             print("hashed %s with %s to be %s" % (name, algorithm, h.hexdigest()))
             return h.hexdigest()
 
-    def replaying(self):
-        with PROXYS[self.proxy](path=self.path, mode="replay") as proxy_service:
-            app_service = APPS[self.app](proxy_service, self.certutil)
-            app_service.start(self.url)
+    def replaying(self, site= None, screen_shot=False):
```
Comment (davehunt) [inline studio.py]: Please run black over your patch before pushing to GitHub. You can install this as a precommit hook. It saves time in reviews so we don't need to discuss issues such as the extra space in `site= None`. You can probably also install black in your IDE.

**Review by davehunt — APPROVED** [review]

**Pull request merged**